src/memory/memory.c

```diff
@@ -710,3 +727,77 @@ bool memory_add_noise_remote_static_pubkey(const uint8_t* pubkey)
 
     return _write_chunk(CHUNK_1, chunk.bytes);
 }
+
+memory_multisig_set_by_hash_result_t memory_multisig_set_by_hash(
+    const uint8_t* hash,
+    const char* name)
+{
+    chunk_2_t chunk = {0};
+    CLEANUP_CHUNK(chunk);
+    _read_chunk(CHUNK_2, chunk_bytes);
+
+    bool found = false;
+    size_t write_index = 0;
+    size_t empty_index = 0;
+    bool empty_found = false;
+
+    uint8_t empty[32];
+    memset(empty, 0xFF, sizeof(empty));
+
+    // Error if there is already a different multisig config with the same name.
+    for (size_t i = 0; i < MEMORY_MULTISIG_NUM_ENTRIES; i++) {
+        const multisig_configuration_t* multisig = &chunk.fields.multisig_configs[i];
+        if (STREQ(multisig->name, name)) {
+            if (!MEMEQ(multisig->hash, hash, sizeof(multisig->hash))) {
+                return MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_DUPLICATE_NAME;
+            }
+            // config already exists (equal), early abort, skipping another write.
+            return MEMORY_MULTISIG_SET_BY_HASH_RESULT_OK;
+        }
+    }
+
+    for (size_t i = 0; i < MEMORY_MULTISIG_NUM_ENTRIES; i++) {
+        const multisig_configuration_t* multisig = &chunk.fields.multisig_configs[i];
+        if (!empty_found && MEMEQ(multisig->hash, empty, sizeof(multisig->hash))) {
+            empty_found = true;
+            empty_index = i;
+        }
+        if (MEMEQ(multisig->hash, hash, sizeof(multisig->hash))) {
+            write_index = i;
+            found = true;
+            break;
+        }
+    }
+    if (!found && !empty_found) {
+        return MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_FULL;
+    }
+    if (!found) {
+        write_index = empty_index;
+    }
+    multisig_configuration_t* multisig = &chunk.fields.multisig_configs[write_index];
+    memcpy(multisig->hash, hash, sizeof(multisig->hash));
+    memset(multisig->name, 0, sizeof(multisig->name));
```

Comment:
It's nicer to use '\0' when emptying a string.

Reply:
Done